### PR TITLE
Added somalier extract for TSO ctDNA BAM

### LINKED ISSUE #509: Add SOMALIER_EXTRACT_STEP to ctTSO500 BAMs for the Holmes pipeline
**Context:**

* Likewise `DRAGEN_WGS_QC` (alignment QC) and `DRAGEN_WTS`; add `SOMALIER_EXTRACT_STEP` upon each `DRAGEN_TSO_CTDNA_STEP` completion.

**Actions:**

* Add into Orchestrator

### PR DESCRIPTION
**Pull request by victorskl**
Fixed #509

### CODE REVIEW

**Review by reisingerf — APPROVED** [review]
LGTM

File: data_processors/pipeline/tools/liborca.py [hunk 1]
```diff
@@ -397,6 +397,42 @@ def parse_transcriptome_output_for_bam_file(workflow_output_json: str, deep_chec
...
+    if deep_check:
+        if output_results_dir_by_sample is None:
+            raise ValueError(f"Unexpected tso_ctdna_tumor_only output. The output_results_dir_by_sample is None")
+
+        if not len(output_results_dir_by_sample) == 1:
+            raise ValueError(f"Unexpected tso_ctdna_tumor_only output. Can't handle multiple samples output")
+
+    results_by_sample = output_results_dir_by_sample[0]
+
+    basename = results_by_sample['basename']  # the main BAM is using <basename>.bam under this directory
```
Comment (alexiswl) [inline data_processors/pipeline/tools/liborca.py:427]: Just wondering what happens if we ever decide to change the name of the output directory from <sample_name> to <sample_name>_tso_outputs or something.  

A slightly more robust way might be to do

```python
listing: List = output_results_dir_by_sample[0]

for listed_item in listing:
	# Only interested in bam files
	if not listed_item.get("nameext", None) == ".bam":
		continue

	# Collect the bam that isn't cleaned.stitched.bam OR the evidence bam
	if listed_item.get("basename", "").startswith("evidence."):
		continue
	if listed_item.get("basename", "").endswith("cleaned.stitched.bam"):
		continue
	
	return listed_item.get("location")
```

Comment (victorskl) [reply data_processors/pipeline/tools/liborca.py:427]: Thanks for checking, Alexis. Sure, cool! I will update and incorporate the proposed BAM grab logic.

Comment (victorskl) [reply data_processors/pipeline/tools/liborca.py]: Updated with https://github.com/umccr/data-portal-apis/pull/518/commits/eb87fab12fb9d4b546d452d034b7bd31642630b0

**Review by alexiswl — APPROVED** [review]

**Pull request merged**